src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/MultipleLastNames.java

```diff
@@ -2,98 +2,74 @@
         // as it won't have been covered by the previous combinations
-        addMultiPartSurnameToCombination(surnameCombinationList, listOfLastNames);
-
-        final List<String> surnameList = new ArrayList<>(surnameCombinationList);
+        addMultiPartLastNameToCombination(lastNameCombinations, largestAllowedName.lastNames());
 
         candidates.addAll(
-                listOfFirstNames.stream()
-                        .flatMap(eachFirstName -> surnameList.stream()
-                                .map(lastNameCombination -> new PersonName(eachFirstName, lastNameCombination)))
+                largestAllowedName.firstNames().stream()
+                        .flatMap(firstName -> lastNameCombinations.stream()
+                                .map(lastNameCombination -> new CandidateName(firstName, lastNameCombination)))
                         .collect(toList())
         );
 
-        addFullNameIfNotAlreadyPresent(candidates, listOfFirstNames, listOfLastNames);
+        addFullNameIfNotAlreadyPresent(candidates, largestAllowedName);
 
         return candidates;
 
     }
 
-    private static boolean multiPart(String lastName) {
-        return lastName.trim().matches(".*\\s+.*");
-    }
+    private static void addMultiPartLastNameToCombination(List<String> lastNameCombinations, List<String> listOfLastNames) {
+        final int MIN_NAMES_FOR_MULTIPART = 3;
 
-    private static void addMultiPartSurnameToCombination(List<String> surnameCombinationList, List<String> listOfLastNames) {
-        if (listOfLastNames.size() > 2) {
-            surnameCombinationList.add(0, StringUtils.join(listOfLastNames, " "));
+        if (listOfLastNames.size() >= MIN_NAMES_FOR_MULTIPART) {
+            lastNameCombinations.add(0, StringUtils.join(listOfLastNames, " "));
         }
     }
 
-    private static void addFullNameIfNotAlreadyPresent(List<PersonName> candidates, List<String> listOfFirstNames, List<String> listOfLastNames) {
-        PersonName fullname = new PersonName(
-                StringUtils.join(listOfFirstNames, " "),
-                StringUtils.join(listOfLastNames, " ")
-        );
-        if (!candidates.contains(fullname)) {
-            candidates.add(0, fullname);
+    private static void addFullNameIfNotAlreadyPresent(List<CandidateName> candidates, InputNames inputNames) {
+        CandidateName fullName = new CandidateName(inputNames.fullFirstName(), inputNames.fullLastName());
+
+        if (!candidates.contains(fullName)) {
+            candidates.add(0, fullName);
         }
     }
 
-    private static List<String> generateSurnameCombinations(List<String> listOfLastNames) {
-        return listOfLastNames.stream()
-                .filter(surname -> surname.length() < HMRC_SURNAME_LENGTH)
+    private static List<String> generateLastNameCombinations(List<String> lastNames) {
+        final Integer HMRC_SURNAME_LENGTH = 3;
```

Comment:
Is there a reason why this is an Integer not an int?

Reply:
I just moved it so I don't know.  I'll change it to an int.